src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2986,16 +2988,19 @@ float Creature::GetAggroRange(Unit const* target) const
     if (aggroRate == 0)
         return 0.0f;
 
-    uint32 targetLevel = target->getLevelForTarget(this);
-    uint32 myLevel = getLevelForTarget(target);
-    int32 levelDiff = int32(targetLevel) - int32(myLevel);
+    uint32 creaturelevel = target->getLevelForTarget(this);
+    uint32 playerlevel  = getLevelForTarget(target);
+    int32 levelDiff = int32(creaturelevel) - int32(playerlevel);
 
     // The maximum Aggro Radius is capped at 45 yards (25 level difference)
     if (levelDiff < -25)
         levelDiff = -25;
 
     // The base aggro radius for mob of same level
-    float aggroRadius = 20.0f;
+    float const detectionRange = GetDetectionRange();
```

Comment:
Isn't this variable kinda useless here? You could simply make:
`float aggroRadius = GetDetectionRange();`

I'm reviewing from smartphone so apologies if this is used below.

Reply:
Theoretically yes , because in the DB there is a default value  20 on detection_range column, but it is always better to have a default value also within the cpp code, even because of a bad value in the DB or an overload via scriptdev. 
```C++ 
float const detectionRange = GetDetectionRange(); > float GetDetectionRange() const { return m_detectionDistance; } > m_detectionDistance(20.0f)
```

Reply:
Maybe I explained myself wrong, I was referring to the variable `detectionRange` not accomplishing anything here. Instead of making this variable, why not simple give `aggroRadius` the default value from function?

Reply:
I see what you mean, but I followed the AC regulations. You'll find a lot of these functions in AC. I won't argue with the work of the AC developers, I guess there's a reason :)

Reply:
I agree with @claudiodfc, I think you can even remove the detectionRange variable and assign to aggroRadius directly `getDetectionRange()`.

you can even use `auto` in this case instead of `float`

Reply:
fixed